app/services/stripe/credit_card_remover.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'stripe/credit_card_clone_destroyer'
+
+module Stripe
+  class CreditCardRemover
+    def initialize(credit_card)
+      @credit_card = credit_card
+    end
+
+    def call
+      Stripe::CreditCardCloneDestroyer.new.destroy_clones(@credit_card)
+
+      stripe_customer = Stripe::Customer.retrieve(@credit_card.gateway_customer_profile_id, {})
+      return unless stripe_customer
+
+      stripe_customer.delete unless stripe_customer.deleted?
```

Comment:
same comment as in the other PR, this could be:
`stripe_customer.delete if stripe_customer && !stripe_customer.deleted?`